docs/articles/extending/multihost-topologies.rst

```diff
@@ -0,0 +1,625 @@
+might say that the test requires "1 client, 1 server and 1 KDC". Or you want to
+test that data replication and load balancer works correctly: "1 client, 3
+servers and 1 load balancer".
+
+The following snippets shows how this can be represented in the configuration
+file:
+
+.. tab-set::
+
+    .. tab-item:: 1 client, 1 server
+
+        .. code-block:: yaml
+
+            domains:
+            - id: myapp
+              hosts:
+              - hostname: client.myapp.test
+                role: client
+              - hostname: server.myapp.test
+                role: server
+
+    .. tab-item:: 1 client, 1 server, 1 KDC
+
+        .. code-block:: yaml
+
+            domains:
+            - id: myapp
+              hosts:
+              - hostname: client.myapp.test
+                role: client
+              - hostname: server.myapp.test
+                role: server
+            - id: authprovider
+              hosts:
+              - hostname: kdc.authprovider.test
+                role: kdc
+
+    .. tab-item:: 1 client, 3 server, 1 load balancer
+
+        .. code-block:: yaml
+
+            domains:
+            - id: myapp
+              hosts:
+              - hostname: client.myapp.test
+                role: client
+              - hostname: server1.myapp.test
+                role: server
+              - hostname: replica1.myapp.test
+                role: server
+              - hostname: replica2.myapp.test
+                role: server
+              - hostname: balancer.myapp.test
+                role: balancer
+
+.. note::
+
+    See that all three servers from the third example are placed inside a single
+    multihost domain. This is because all these servers contains the same data
+    and it should not matter to which one the client talks to. If they serve
+    different data, they should be placed in different multihost domains.
+
+Topology Marker
+===============
+
+Pytest-mh implements a new marker ``@pytest.mark.topology`` which is converted
+into an instance of :class:`~pytest_mh.TopologyMark`. This marker is used to
+assign a topology to a test. One test can be associated with multiple topologies
+-- this is called :ref:`topology parametrization <topology_parametrization>`. In
+this case, the test is multiplied and run once for all assigned topologies,
+therefore it is possible to re-use the test code for different setups/backends.
+
+The ``@pytest.mark.topology`` can take different types of arguments in order to
+instantiate the marker:
+
+* individual arguments, see :ref:`adhoc_topology`
+* single instance or list of :class:`~pytest_mh.TopologyMark`, see :ref:`predefined_topology`
+* values from :class:`~pytest_mh.KnownTopologyBase` or
+  :class:`~pytest_mh.KnownTopologyGroupBase` enums (**recommended**), see
+  :ref:`known_topology`
+
+.. _adhoc_topology:
+
+Ad-hoc Topologies
+-----------------
+
+If you plan to use the topology only once, you can define it directly on the
```

Comment:
Why would you use an ad hoc topology? I don't understand the reason/use case.

Reply:
For one-time use, but in the end there is note that pre-defined topologies should be preferred.